include/alpaka/dev/DevUniformCudaHipRt.hpp

```diff
@@ -57,6 +57,10 @@ namespace alpaka
         DevUniformCudaHipRt() = default;
 
     public:
+        DevUniformCudaHipRt(int iDevice) : m_iDevice(iDevice)
```

Comment:
This allows the creation of a `DevUniformCudaHipRt` with an arbitrary device id, even an invalid one.
_cf._ [`PltfUniformCudaHipRt::GetDevByIdx::getDevByIdx()`](https://github.com/alpaka-group/alpaka/blob/2d57d41f076735520d17d077d22469b8fe6842f7/include/alpaka/pltf/PltfUniformCudaHipRt.hpp#L69-L110)

Reply:
you could make this constructor `private` and `PltfUniformCudaHipRt::GetDevByIdx::getDevByIdx()` a `friend`